src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/Signatures/SignatureHeader.cs

```diff
@@ -40,7 +40,8 @@ public SignatureCallingConvention CallingConvention
             {
                 int callingConventionOrKind = _rawValue & CallingConventionOrKindMask;
 
-                if (callingConventionOrKind > maxCallingConvention)
+                if (callingConventionOrKind > maxCallingConvention
+                    && callingConventionOrKind != (int)SignatureCallingConvention.Unmanaged)
```

Comment:
Add similar change to `SignatureKind Kind` as well?

Reply:
Is there a good reason for making these values non-contiguous?

Reply:
Yes, these calling convention values share the encoding space with other signature types in the ECMA-335 file format.

Reply:
Should we make all of them part of this enum?  It seems very usual to me that we'd have non-public values of data that is stored in the metadata.

Reply:
The other values are exposed as SignatureKind enum already: https://docs.microsoft.com/en-us/dotnet/api/system.reflection.metadata.signaturekind?view=netcore-3.1

System.Reflection.Metadata object model uses the two different enums for the same encoding spaces as guiderails.